tests/rptest/tests/full_disk_test.py

```diff
@@ -130,6 +138,67 @@ def test_write_rejection(self):
                 expect_blocked=True)
             self.full_disk.clear_low_space()
 
+    @cluster(num_nodes=3, log_allow_list=FDT_LOG_ALLOW_LIST)
+    def test_reset_health_monitor(self):
+        """ Verify that health monitor frontend and backend have valid state after
+        they are reset by writing enough data that the disk health check triggers.
+        """
+        def check_health_monitor_frontend(disk_space_change: str):
+            # Looking for a log statement about a change in disk space.
+            # This is a check for the health monitor frontend because
+            # that structure logs disk space alerts.
+            pattern = f"Update disk health cache {disk_space_change}"
+            wait_until(
+                lambda: self.redpanda.search_log_any(pattern),
+                timeout_sec=5,
+                err_msg=f"Failed to find disk space change: {disk_space_change}"
+            )
```

Comment:
nit: this pattern is similar to `check_service_restart`. Consider renaming this to something more general and moving it to the utils.py file

Reply:
I put this in a separate branch in https://github.com/NyaliaLui/redpanda/commit/770b5104917bd532d856bd43d0401d43867c802c but the PR for it will come after this PR merges and after #9105 merges.